src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/CpsPackageReferenceProject.cs

```diff
@@ -317,9 +317,7 @@ public override async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsy
             BuildIntegratedInstallationContext installationContext,
             CancellationToken token)
         {
-            // Right now, the UI only handles installation of specific versions, which is just the minimum version of
-            // the provided version range.
-            var formattedRange = range.MinVersion.ToNormalizedString();
+            var formattedRange = range.OriginalString ?? range.MinVersion.ToNormalizedString();
```

Comment:
Have you tried out legacy PackageReference projects?

You might need to make a similar change there.

Reply:
@nkolev92 I tried and it looks that Legacy already handles custom versions